diff --git a/android/build.gradle b/android/build.gradle
index f324b9a9f3c3b3cf28eaac7fa09471592212bf3f..8b8170d4f3daa68e33342acc0a27b4943fd62894 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -8,7 +8,7 @@ buildscript {
 
     dependencies {
         classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version")
-        classpath("com.android.tools.build:gradle:7.3.1")
+        classpath("com.android.tools.build:gradle:8.12.3")
         classpath("com.diffplug.spotless:spotless-plugin-gradle:6.11.0")
     }
 }
@@ -47,7 +47,7 @@ android {
         }
     }
 
-    compileSdkVersion getExtOrDefault('compileSdkVersion', 30)
+    compileSdkVersion 36
 
     // Used to override the NDK path/version on internal CI or by allowing
     // users to customize the NDK path/version from their root project (e.g. for M1 support)
@@ -59,8 +59,8 @@ android {
     }
 
     defaultConfig {
-        minSdkVersion getExtOrDefault('minSdkVersion', 16)
-        targetSdkVersion getExtOrDefault('targetSdkVersion', 28)
+        minSdkVersion getExtOrDefault('minSdkVersion', 21)
+        targetSdkVersion getExtOrDefault('targetSdkVersion', 36)
         versionCode 1
         versionName "1.0"
         buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
diff --git a/android/gradle.properties b/android/gradle.properties
index ce9ec6994204558b0af0d68c02010b80019c3565..3454d8aa42a194b18c89e07bbc825d297466a051 100644
--- a/android/gradle.properties
+++ b/android/gradle.properties
@@ -1 +1 @@
-RNSAC_kotlinVersion=1.6.20
+RNSAC_kotlinVersion=2.2.20
